### PR TITLE
Fix Management Group CreateUpdate Function

### PR DESCRIPTION
**Pull request by wsf11**
Addresses issue #6091 

Issue: The Azure REST API returns a 403 if either you do not have access to a management group or a management group cannot be found. When creating a new management group, a check is always performed to see if the management group exists already, causing a 403, and an error. 

Fix: Change response not found to response forbidden and log message. In addition, there was a regression in PR #6276 that did not check for the name, rather it was checking for the group_name

### CODE REVIEW

**Comment by wsf11** [comment]
@katbyte Let me know if you have any feedback on this PR!

File: azurerm/utils/response.go [hunk 1]
```diff
@@ -36,3 +36,7 @@ func ResponseWasStatusCode(resp autorest.Response, statusCode int) bool { // nol
 
 	return false
 }
+
+func ResponseWasForbidden(resp autorest.Response) bool {
```
Comment (mbfrahry) [inline azurerm/utils/response.go:40]: It looks like this was added recently so we can remove it.

**Comment by jackofallops** [comment]
Quick update - Our CI System still being worked on to be able to run the suite there to be able to merge/release this, but we're getting there... When running locally:

Tests Pass
```
--- PASS: TestAccAzureRMManagementGroup_basic (54.19s)
--- PASS: TestAccAzureRMManagementGroup_requiresImport (58.97s)
--- PASS: TestAccAzureRMManagementGroup_withName (62.22s)
--- PASS: TestAccDataSourceArmManagementGroup_basic (65.46s)
--- PASS: TestAccAzureRMManagementGroup_withSubscriptions (85.23s)
--- PASS: TestAccAzureRMManagementGroup_nested (88.34s)
--- PASS: TestAccAzureRMManagementGroup_multiLevel (121.81s)
--- PASS: TestAccAzureRMManagementGroup_multiLevelUpdated (147.49s)
PASS
ok  	github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/managementgroup/tests	227.362s
```

**Comment by jackofallops** [comment]
I've looped back around to this now I can test review it properly.
What I've found is that the service appears to respond with a 403 when a `get` is sent for a non-existent group ID, where it _should_ respond with a 404. This behaviour would have been masked pre-2.0 as there was a conditional test on the import / existence check (`features.ShouldResourcesBeImported()`) so unless opted into that feature in 1.x, one would never see it.
As a result, I think this needs a slightly different change for this bug. I'll suggest in a review for that.

There's also the name update bug as mentioned by @ArcturusZhang - But I think that will need some careful thought and probably separating the Create and Update functions, as the change of name appears to change the resource ID of the group also.

**Comment by jackofallops** [comment]
Hi @wsf11 - I hope you don't mind, I've pushed a few changes to address the issue and resolve a couple other items.  Tests are now passing fully:
image
image

403 issue logged [here](https://github.com/Azure/azure-rest-api-specs/issues/9549)

**Comment by wsf11** [comment]
@jackofallops no worries. Glad I could help with this!

**Pull request merged**